crates/fluvio-version-manager/src/command/switch.rs

```diff
@@ -0,0 +1,221 @@
+//! Version Switching Command
+//!
+//! The `switch` command is responsible of changing the active Fluvio Version
+
+use std::fs::{create_dir_all, read_dir, copy};
+use std::path::{PathBuf, Path};
+
+use anyhow::Result;
+use clap::Parser;
+use colored::Colorize;
+
+use fluvio_hub_util::fvm::Channel;
+
+use crate::GlobalOptions;
+use crate::common::manifest::{PACKAGE_SET_MANIFEST_FILENAME, VersionManifest};
+use crate::common::settings::Settings;
+use crate::common::workdir::{fvm_versions_path, fluvio_binaries_path};
+use crate::common::notify::Notify;
+
+/// Represents the contents of a version directory (`~/.fvm/versions/<version>`)
+/// where binaries and the manifest are stored.
+///
+/// The `VersionDirectory` represents the contents of a version directory
+/// in 2 parts:
+///
+/// - `contents`: Every file in the directory except for the manifest
+/// - `manifest`: The manifest file
+struct VersionDirectory {
+    contents: Vec<PathBuf>,
+    manifest: PathBuf,
+}
+
+impl VersionDirectory {
+    /// Opens a directory and returns a `VersionDirectory` build from scanning it
+    fn open(path: PathBuf) -> Result<Self> {
+        let mut dir = Self {
+            contents: vec![],
+            manifest: PathBuf::new(),
+        };
+
+        for entry in read_dir(path)? {
+            let entry = entry?;
+
+            if entry.metadata()?.is_dir() {
+                continue;
+            }
+
+            let entry_path = entry.path();
+
+            if entry_path.file_name().ok_or(anyhow::anyhow!(
+                "Failed to get filename from path: {}",
+                entry_path.display()
+            ))? == PACKAGE_SET_MANIFEST_FILENAME
+            {
+                dir.manifest = entry_path;
+            } else {
+                dir.contents.push(entry_path);
+            }
+        }
+
+        if dir.manifest == PathBuf::new() {
+            return Err(anyhow::anyhow!(
+                "Failed to find manifest file in version directory"
+            ));
+        }
+
+        Ok(dir)
+    }
+
+    /// Copies `contents` files into the `target` directory
+    fn copy_contents(&self, target: impl AsRef<Path>) -> Result<()> {
+        for entry in &self.contents {
+            let filename = entry.file_name().ok_or(anyhow::anyhow!(
+                "Failed to get filename from path: {}",
+                entry.display()
+            ))?;
+            let target_path = target.as_ref().join(filename);
+
+            copy(entry, &target_path)?;
+            tracing::info!(?target_path, "Copied binary");
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Debug, Parser)]
+pub struct SwitchOpt {
+    #[command(flatten)]
+    global_opts: GlobalOptions,
+    /// Version to install
+    #[arg(index = 1, default_value_t = Channel::Stable)]
+    version: Channel,
```

Comment:
If the user runs `fvm switch` with no params, it will switch to Stable by default. WDYT about this behavior?

Reply:
I think `fvm switch` with no params should be an error.  Maybe it could print an error string and recommend running `fvm show` to see versions available to switch.

Reply:
Agree!

Reply:
Just remove default

Reply:
I added the following behavior here: [3272fd5](https://github.com/infinyon/fluvio/pull/3597/commits/3272fd5aff1a751d16b21f0ed7203f681d3102bd)

- Shows a help message to recommend using the `fvm show` command
- Exits with code `1` and the message: `Error: No version provided`

<img width="760" alt="Screenshot 2023-10-12 at 5 51 31 PM" src="https://github.com/infinyon/fluvio/assets/34756077/94bc510d-cfef-46d0-b327-ef00c060d2bb">